src/DotVVM.Framework/Controls/UpdateProgress.cs

```diff
@@ -21,6 +36,11 @@ protected override void AddAttributesToRender(IHtmlWriter writer, IDotvvmRequest
         {
             writer.AddKnockoutDataBind("dotvvm-UpdateProgress-Visible", "true");
 
+            if (Delay != 0)
```

Comment:
Ok, But that's somehow strange that if you specify 0 ms delay you get **NO** delay and if you specify negative delay, for example -1, you get the minimum delay (**4 ms**).